src/main/java/cn/nukkit/entity/Entity.java

```diff
@@ -596,15 +596,26 @@ protected final void init(FullChunk chunk, CompoundTag nbt) {
         this.setDataProperty(new FloatEntityData(DATA_SCALE, scale), false);
         this.setDataProperty(new ByteEntityData(DATA_COLOR, 0), false);
 
-        this.chunk.addEntity(this);
-        this.level.addEntity(this);
+        try {
+            this.chunk.addEntity(this);
+            this.level.addEntity(this);
 
-        this.initEntity();
+            this.initEntity();
 
-        this.lastUpdate = this.server.getTick();
-        this.server.getPluginManager().callEvent(new EntitySpawnEvent(this));
+            this.lastUpdate = this.server.getTick();
 
-        this.scheduleUpdate();
+            EntitySpawnEvent event = new EntitySpawnEvent(this);
+
+            this.server.getPluginManager().callEvent(event);
+
+            if (event.isCancelled()) {
+                this.close(false);
+            } else {
+                this.scheduleUpdate();
+            }
+        } catch(Exception e) {
+            this.close(false);
```

Comment:
Could you add some console log into this point?

Reply:
I think it's better to re-throw the exception instead

Reply:
that would let the caller handle the exception

Reply:
Yes, thank you for reminding me of this.

Reply:
I agree